libflux/src/core/semantic/mod.rs

```diff
@@ -82,3 +90,25 @@ pub fn convert_source(source: &str) -> Result<nodes::Package, Error> {
     )?;
     Ok(nodes::inject_pkg_types(sem_pkg, &sub))
 }
+
+/// Given a Flux source and a variable name, find out the type of that variable in the Flux source code.
+/// A type variable will be automatically generated and injected into the type environment that
+/// will be used in semantic analysis. The Flux source code itself should not contain any definition
+/// for that variable.
+pub fn find_var_type(source: &str, var_name: &str) -> Result<MonoType, Error> {
```

Comment:
Should this take an optional type environment so that we don't have to duplicate the same code in the `flux` package?

Reply:
The only overlap of those two `find_var_type()` functions is injecting type var into an environment. The fresher, env, built-ins, importer, almost every parameter is prepared differently. I don't really like to add many parameters to reuse just one environment.add() logic.

Reply:
Ok. Is there an argument for keeping both functions?